packages/core/utils/src/modules-sdk/event-builder-factory.ts

```diff
@@ -0,0 +1,44 @@
+import { Context, EventBusTypes } from "@medusajs/types"
+import { Modules } from "./definition"
+
+export function eventBuilderFactory({
+  isMainEntity,
+  action,
+  object,
+  eventsEnum,
+}: {
+  isMainEntity?: boolean
+  action: string
+  object: string
+  eventsEnum: Record<string, string>
+}) {
+  return function ({
+    data,
+    sharedContext,
+  }: {
+    data: { id: string }[]
+    sharedContext: Context
+  }) {
+    if (!data.length) {
+      return
+    }
+
+    const aggregator = sharedContext.messageAggregator!
+    const messages: EventBusTypes.RawMessageFormat[] = []
+
+    data.forEach((dataItem) => {
+      messages.push({
+        service: Modules.FULFILLMENT,
```

Comment:
This will need to be passed as an argument

Reply:
done 😂